terraform-provider-constellation/examples/full/azure_cluster.tf

```
@@ -0,0 +1,111 @@
+terraform {
+  required_providers {
+    constellation = {
+      source  = "edgelesssys/constellation"
+      version = "X.Y.Z"
+    }
+    random = {
+      source  = "hashicorp/random"
+      version = "3.6.0"
+    }
+  }
+}
+
+locals {
+  name                = "constell"
+  version             = "vX.Y.Z"
+  csp                 = "azure"
+  attestation_variant = "azure-sev-snp"
+  location            = "northeurope"
+
+  master_secret      = random_bytes.master_secret.hex
+  master_secret_salt = random_bytes.master_secret_salt.hex
+  measurement_salt   = random_bytes.measurement_salt.hex
+}
+
+resource "random_bytes" "master_secret" {
+  length = 32
+}
+
+resource "random_bytes" "master_secret_salt" {
+  length = 32
+}
+
+resource "random_bytes" "measurement_salt" {
+  length = 32
+}
+
+module "azure_iam" {
+  source                 = "./infrastructure/iam/azure" # TODO: Replace with remote source URL
+  region                 = local.location
+  service_principal_name = "${local.name}-test-sp"
+  resource_group_name    = "${local.name}-test-rg"
+}
+
+module "azure_infrastructure" {
+  source                 = "./infrastructure/azure" # TODO: Replace with remote source URL
+  name                   = local.name
+  user_assigned_identity = module.azure_iam.uami_id
+  node_groups = {
+    control_plane_default = {
+      role          = "control-plane"
+      instance_type = "Standard_DC4as_v5"
+      disk_size     = 30
+      disk_type     = "Premium_LRS"
+      initial_count = 3
+    },
+    worker_default = {
+      role          = "worker"
+      instance_type = "Standard_DC4as_v5"
+      disk_size     = 30
+      disk_type     = "Premium_LRS"
+      initial_count = 2
+    }
+  }
+  location       = local.location
+  image_id       = data.constellation_image.bar.reference
+  resource_group = module.azure_iam.base_resource_group
+  create_maa     = true
+}
+
+data "constellation_attestation" "foo" {
+  csp                 = local.csp
+  attestation_variant = local.attestation_variant
+  image_version       = local.version
+}
```

```suggestion
data "constellation_attestation" "foo" {
  csp                 = local.csp
  attestation_variant = local.attestation_variant
  image_version       = local.version
  maa_url             = module.azure_infrastructure.maa_url
}
```
For completeness, we should add the MAA URL to the Azure example.
(Variable reference might be incorrect, not sure what outputs are defined for the module)